Include sbt-ivy-snapshots to launchconfig

eed3si9n opened the pull request:
## what this changes

both `typesafe-ivy-releases` and `sbt-ivy-snapshots` repos are always included into the launchconfig.
## notes

Prior to this change, the launchconfig for sbt was generated using Transform's resource generator so releases would include `typesafe-ivy-releases` repo, and snapshot releases would include `typesafe-ivy-snapshots` and `sonatype-snapshots` repos in addition. This is no longer useful since nightly builds are now published to `sbt-ivy-snapshots` instead, which is backed by Bintray.

This removes the switching logic, so both `typesafe-ivy-releases` and `sbt-ivy-snapshots` repos are always included into the launchconfig. So it's no longer necessary to download a launcher just to try the nightly builds.

/review @dwijnand, @Duhemm, @jsuereth

dwijnand commented:
Excellent. Thanks. LGTM.

dwijnand commented:
FYI Out of interest and as a sort of smoke/sanity check I made this change to the 0.13.10-20151229-205306 launcher I had downloaded, and it downloaded all the modules, and it downloaded the relevant bridge jars to compiler the compiler-interface (to compile project/foo.scala).

eed3si9n commented:
It seems like I went too far with removing the resource gen. Fix coming up...

```
Running actions / add-alias
[info] Getting ${{org}} sbt ${{sbt.version}} ...
[info] 
[info] :: problems summary ::
[info] :::: WARNINGS
[info]      module not found: ${{org}}#sbt;${{sbt.version}}
[info] 
[info]  ==== local: tried
[info] 
[info]    /home/travis/.ivy2/local/${{org}}/sbt/${{sbt.version}}/ivys/ivy.xml
[info] 
[info]    -- artifact ${{org}}#sbt;${{sbt.version}}!sbt.jar:
[info] 
[info]    /home/travis/.ivy2/local/${{org}}/sbt/${{sbt.version}}/jars/sbt.jar
[info] 
[info]  ==== Maven Central: tried
[info] 
[info]    https://repo1.maven.org/maven2/${{org}}/sbt/${{sbt.version}}/sbt-${{sbt.version}}.pom
[info] 
[info]    -- artifact ${{org}}#sbt;${{sbt.version}}!sbt.jar:
[info] 
[info]    https://repo1.maven.org/maven2/${{org}}/sbt/${{sbt.version}}/sbt-${{sbt.version}}.jar
[info] 
[info]  ==== typesafe-ivy-releases: tried
[info] 
[info]    https://repo.typesafe.com/typesafe/ivy-releases/${{org}}/sbt/${{sbt.version}}/ivys/ivy.xml
[info] 
[info]  ==== sbt-ivy-snapshots: tried
[info] 
[info]    https://repo.scala-sbt.org/scalasbt/ivy-snapshots/${{org}}/sbt/${{sbt.version}}/ivys/ivy.xml
[info] 
[info]      ::::::::::::::::::::::::::::::::::::::::::::::
[info] 
[info]      ::          UNRESOLVED DEPENDENCIES         ::
[info] 
[info]      ::::::::::::::::::::::::::::::::::::::::::::::
[info] 
[info]      :: ${{org}}#sbt;${{sbt.version}}: not found
[info] 
[info]      ::::::::::::::::::::::::::::::::::::::::::::::
[info] 
```

eed3si9n committed:
Include sbt-ivy-snapshots to launchconfig

Include sbt-ivy-snapshots to launchconfig
Prior to this change, the launchconfig for sbt was generated using
Transform's resource generator so releases would include
`typesafe-ivy-releases` repo, and snapshot releases would include
`typesafe-ivy-snapshots` and `sonatype-snapshots` repos in addition.
This is no longer useful since nightly builds are now published to
`sbt-ivy-snapshots` instead, which is backed by Bintray.

This removes the switching logic, so both `typesafe-ivy-releases` and
`sbt-ivy-snapshots` repos are always included into the launchconfig.
This removes the step of needing to download a launcher just to try the
nightly builds.
